File: LeetCode/python/lc_0001.py
# -*- coding: utf-8 -*-
"""
 * @file     twoSum.py
 * @brief    
 * @author   YongDu
 * @date     2021-07-25
"""


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = dict()
        for i, num in enumerate(nums):
            if target - num in hash_map:
                return [hash_map[target - num], i]
            hash_map[nums[i]] = i

        return []

File: LeetCode/cpp/lc_0001.cpp
/**
 * @file     lc_0001.cpp
 * @brief    https://leetcode-cn.com/problems/two-sum/
 * @author   YongDu
 * @date     2021-09-08
 */
class Solution {
  public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> indexMap;
        for (int i = 0; i < nums.size(); ++i) {
            if (indexMap.find(target - nums[i]) != indexMap.end()) {
                return std::vector{i, indexMap[target - nums[i]]};
            } else {
                indexMap[nums[i]] = i;
            }
        }
        return std::vector<int>();
    }
};
